splink/case_statements.py

```diff
@@ -274,4 +318,237 @@ def sql_gen_gammas_name_inversion_4(col_name:str, other_name_cols:list, gamma_co
     if gamma_col_name is not None:
         return _add_as_gamma_to_case_statement(c, gamma_col_name)
     else:
-        return c
\ No newline at end of file
+        return c
+
+
+# Array comparison functions
+
+
+def sql_gen_case_stmt_array_intersect_2(
+    col_name: str, gamma_col_name=None, zero_length_is_null=True
+):
+    """Generate a case comparison which is 1 if the size of the intersection of the arrays is
+    one or more.  Otherwise zero.
+
+
+    Args:
+        col_name (str): The name of the column we want to generate a custom case expression for e.g. phone_number
+        gamma_col_name (str, optional): . The name of the column, for the alias e.g. surname
+        zero_length_is_null (bool, optional):  Whether to treat a zero length array as a null. Defaults to True.
+    """
+
+    zero_length_expr = ""
+    if zero_length_is_null:
+        zero_length_expr = (
+            f"when size({col_name}_l) = 0 or size({col_name}_r) = 0 then -1"
```

Comment:
I have used `fix_zero_length_arrays` from `splink_data_standardisation.arrays` to replace empty arrays will null. In that case `size` returns -1 rather than 0. This is why I used `< 1` rather than `=0` here.

Reply:
Just realised this isn't necessary because it's already preceded by the separate line checking for `null`